lib/std/spinlock.zig

```diff
@@ -20,9 +22,56 @@ pub const SpinLock = struct {
     }
 
     pub fn acquire(self: *SpinLock) Held {
-        while (@atomicRmw(u8, &self.lock, builtin.AtomicRmwOp.Xchg, 1, AtomicOrder.SeqCst) != 0) {}
+        var backoff = Backoff.init();
+        while (@atomicRmw(u8, &self.lock, .Xchg, 1, .Acquire) != 0)
+            backoff.yield();
         return Held{ .spinlock = self };
     }
+
+    pub fn yieldCpu() void {
+        switch (builtin.arch) {
+            .i386, .x86_64 => asm volatile("pause" ::: "memory"),
+            // .arm, .aarch64 => asm volatile("yield"),
+            //
+            // Causes CI to fail
+            // See: https://github.com/ziglang/zig/pull/3585#issuecomment-549962765
+            else => time.sleep(0),
+        }
+    }
+
+    pub fn yieldThread() void {
+        switch (builtin.os) {
+            .linux => assert(linux.syscall0(linux.SYS_sched_yield) == 0),
```

Comment:
Would you mind adding the definition for `sched_yield` in std/os/linux.zig, and the corresponding code in std/os.zig? This will accomplish several things:

 * Improve the std lib to have `std.os.linux.sched_yield`
 * Improve the std lib to have `std.os.sched_yield`, which works on POSIX systems that link libc. 
 * Make `std.os.sched_yield` call the libc function on linux, if libc is being linked. (This is accomplished by following the pattern of the other functions in that file, and usually does not need explicit `if (builtin.link_libc) ...` logic.)
 * At this callsite it will be nicer since the return type will be `void`.

Since `sched_yield` appears to be a POSIX function, this would mean you only need the logic here for windows; once the above changes are done, all other systems could do `os.sched_yield()`.

In fact, it would be reasonable to put `kernel32.SwitchToThread()` in the windows impl of `std.os.sched_yield`, and then this entire `yieldThread` function would simply call `std.os.sched_yield`. (At that point it becomes implementor's choice whether to even have this `yieldThread()` function, or update the callsites to use std.os.sched_yield directly)